### PR TITLE
Get Team Custom Fields, Set User Custom Fields

### PR DESCRIPTION
**Pull request by MattDavisRV**
Added functionality to retrieve defined custom fields from the the Team.  

Added ability to update User custom fields from the information available from the new team request.

### CODE REVIEW

**Comment by MattDavisRV** [comment]
Slack separates Profile and Custom Field update requests.  This was needed to update additional fields that were not apart of the main profile request.  To issue the update request, slack doesn't use a conventional request structure, requiring the need for the map[string]UserCustomProfile. 

Requesting the team custom variables was added to know which IDs to send in the above request. 

Open to suggestions and/or updates to this PR.

File: users.go [hunk 1]
```diff
@@ -464,6 +464,56 @@ func (api *Client) DeleteUserPhotoContext(ctx context.Context) error {
...
 
+// SetUserCustomFields sets Custom Profile fields on the provided users account.  Due to the non-repeating elements
+// within the request, a map fields is required.  The key in the map signifies the field that will be updated.
+//
+// See GetTeamProfile for information to retrieve possible fields for your account.
+func (api *Client) SetUserCustomFields(userID string, customFields map[string]UserProfileCustomField) error {
+	return api.SetUserCustomFieldsContext(context.Background(), userID, customFields)
+}
+
+// SetUserCustomFieldsContext will set a custom status and emoji for the currently authenticated user with a custom context
```
Comment (olivierlemasle) [inline users.go]: Comment is wrong (copy-paste error) ;-)

**Review by olivierlemasle — APPROVED** [review]
:+1:

File: users.go (hunk 1 above)
Comment (MattDavisRV) [reply users.go]: Good catch!  Should be fine now

**Comment by james-lawrence** [comment]
TeamProfile.... not familiar with this part of slacks api. does it represent customization values for the team itself? not the users?

**Comment by MattDavisRV** [comment]
TeamProfile defines custom fields that belong to the team, but attributed to the user.  They're represented as a ID/Value pair, which can only be retrieved from the added code. 

As an example (in my use case), I've added a custom field called employeeId to the team, then added this attribute to each user to properly identify them during interactive messages (dialogs, etc).

File: users.go [hunk 2]
```diff
@@ -464,6 +464,56 @@ func (api *Client) DeleteUserPhotoContext(ctx context.Context) error {
...
+		"user":    {userID},
+		"profile": {string(profile)},
+	}
+
+	response := &userResponseFull{}
+	if err := postForm(ctx, api.httpclient, APIURL+"users.profile.set", values, response, api); err != nil {
+		return err
+	}
+
+	if !response.Ok {
```
Comment (james-lawrence) [inline users.go:509]: return response.Err()

**Review by james-lawrence — CHANGES_REQUESTED** [review]
one minor comment.

File: users.go (hunk 2 above)
Comment (james-lawrence) [inline users.go:509]: sorry if i wasnt clear. response.Err() does `if !response.Ok {}`

you can literally just say `reutnr response.Err()` here

**Comment by olivierlemasle** [comment]
@MattDavisRV You removed the branch `SetUserCustomFields` from your fork repo. Was it on purpose?

**Comment by MattDavisRV** [comment]
@olivierlemasle no, it was not on purpose.  Not really sure what happened there.  I've been busy with work.  I'll try to reimplement since it was a small change along with the change requested here.

**Comment by MattDavisRV** [comment]
I'm waiting for #496 to be completed before I swing back to this PR.  I will complete the requested changes then.

**Pull request closed without merging**